pex/pep_376.py

```diff
@@ -0,0 +1,430 @@
+
+        # The RECORD is a csv file with the path to each installed file in the 1st column.
+        # See: https://www.python.org/dev/peps/pep-0376/#record
+        with closing(
+            fileinput.input(files=[record_abspath], inplace=True, mode=MODE_READ_UNIVERSAL_NEWLINES)
+        ) as record_fi:
+            csv_writer = None  # type: Optional[CSVWriter]
+            for installed_file in Record.read(record_fi):
+                if csv_writer is None:
+                    # N.B.: The raw input lines include a newline that varies between '\r\n' and
+                    # '\n' when the wheel was built from an sdist by Pip depending on whether the
+                    # interpreter used was Python 2 or Python 3 respectively. As such, we normalize
+                    # all RECORD files to use '\n' regardless of interpreter.
+                    csv_writer = cast(
+                        "CSVWriter",
+                        csv.writer(sys.stdout, delimiter=",", quotechar='"', lineterminator="\n"),
+                    )
+
+                abspath_to_rehash = to_rehash.pop(installed_file.path, None)
+                if installed_file.hash and abspath_to_rehash is not None:
+                    digest = installed_file.hash.parse()
+                    hasher = digest.new_hasher()
+                    with open(abspath_to_rehash, "rb") as rehash_fp:
+                        CacheHelper.update_hash(rehash_fp, digest=hasher)
+
+                    fingerprint = base64.urlsafe_b64encode(hasher.digest()).decode("ascii")
+                    de_padded, pad, rest = fingerprint.rpartition("=")
+                    new_hash = str(de_padded if pad and not rest else fingerprint)
+                    new_size = os.stat(abspath_to_rehash).st_size
+                    csv_writer.writerow(
+                        (
+                            installed_file.path,
+                            "{alg}={hash}".format(alg=digest.algorithm, hash=new_hash),
+                            new_size,
+                        )
+                    )
+                elif installed_file.path != direct_url_relpath:
+                    csv_writer.writerow(
+                        (
+                            installed_file.path,
+                            str(installed_file.hash) if installed_file.hash is not None else "",
+                            str(installed_file.size) if installed_file.size is not None else "",
+                        )
+                    )
+
+    def reinstall(
+        self,
+        venv,  # type: Virtualenv
+        exclude=None,  # type: Optional[Callable[[str], bool]]
+        symlink=False,  # type: bool
+        rel_extra_path=None,  # type: Optional[str]
+    ):
+        # type: (...) -> Iterator[Tuple[str, str]]
+        """Re-installs the installed wheel in a venv.
+
+        N.B.: A record of reinstalled files is returned in the form of an iterator that must be
+        consumed to drive the installation to completetion.
```

Comment:
```suggestion
        consumed to drive the installation to completion.
```

Reply:
I'll try to get this in a follow-up that touches this file. I routinely pass on typo fixes in non-public docs since it burns trees and the proof the typo is not severe is already there in the fact it could be understood enough to be corrected.

Reply:
Yeah, I don't think typos are worth a CI round, but can be incidentally fixed in other changes.